lib/datadog/tracing/distributed/trace_context.rb

```diff
@@ -299,10 +304,14 @@ def extract_datadog_fields(dd_tracestate)
 
               tags ||= {}
               tags["#{Tracing::Metadata::Ext::Distributed::TAGS_PREFIX}#{key}"] = value
+            else
+              unknown_fields ||= String.new
+              unknown_fields << pair
+              unknown_fields << ';'
             end
           end
 
-          [origin, sampling_priority, tags]
+          [origin, sampling_priority, tags, unknown_fields]
```

Comment:
Perhaps we should extract this to its own object, if the array keeps on growing. It feels weird to me that the caller need to understand the index of the array to get the corresponding value.

Reply:
I'll do it if we get to 5! 🦆